engine-upgrade-utils/src/lib.rs

```diff
@@ -142,5 +151,11 @@ fn test_c_str_array_with_args() {
 	let args = vec!["arg1".to_string(), "arg2".to_string()];
 
 	let c_args: CStrArray = args.clone().try_into().unwrap();
+	// check the Clone/drop implementations
+	{
+		let c_args_2: CStrArray = c_args.clone();
+		drop(c_args_2);
```

Comment:
Does this actually check anything? Don't we have to check memory or something?

Reply:
Yes, before this commit's change, this panics, it was doing a double free